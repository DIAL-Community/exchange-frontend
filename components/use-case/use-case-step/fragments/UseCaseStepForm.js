import React, { useCallback, useContext, useState } from 'react'
import { useRouter } from 'next/router'
import { Controller, useForm } from 'react-hook-form'
import { FaSpinner } from 'react-icons/fa6'
import { useIntl } from 'react-intl'
import { useMutation } from '@apollo/client'
import { ToastContext } from '../../../../lib/ToastContext'
import { HtmlEditor } from '../../../shared/form/HtmlEditor'
import Input from '../../../shared/form/Input'
import ValidationError from '../../../shared/form/ValidationError'
import { CREATE_USE_CASE_STEP } from '../../../shared/mutation/useCaseStep'

const UseCaseStepForm = React.memo(({ useCaseStep, useCase }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const router = useRouter()

  const [mutating, setMutating] = useState(false)
  const [reverting, setReverting] = useState(false)

  const { showSuccessMessage, showFailureMessage } = useContext(ToastContext)
  const { locale } = useRouter()

  const [updateUseCaseStep, { reset }] = useMutation(CREATE_USE_CASE_STEP, {
    onError: () => {
      showFailureMessage(format('toast.submit.failure', { entity: format('ui.useCaseStep.label') }))
      setMutating(false)
      reset()
    },
    onCompleted: (data) => {
      const { createUseCaseStep: response } = data
      if (response?.useCaseStep && response?.errors?.length === 0) {
        setMutating(false)
        showSuccessMessage(
          format('toast.submit.success', { entity: format('ui.useCaseStep.label') }),
          () => router.push(
            `/${locale}` +
            `/use-cases/${data.createUseCaseStep.useCaseStep.useCase.slug}` +
            `/use-case-steps/${data.createUseCaseStep.useCaseStep.slug}`
          )
        )
      } else {
        showFailureMessage(format('toast.submit.failure', { entity: format('ui.useCaseStep.label') }))
        setMutating(false)
        reset()
      }
    }
  })

  const { handleSubmit, register, control, formState: { errors } } = useForm({
    mode: 'onSubmit',
    reValidateMode: 'onChange',
    shouldUnregister: true,
    defaultValues: {
      name: useCaseStep?.name,
      stepNumber: useCaseStep?.stepNumber,
      description: useCaseStep?.useCaseStepDescription?.description
    }
  })

  const slug = useCaseStep?.slug ?? ''
  const useCaseId = parseInt(useCase?.id)

  const doUpsert = async (data) => {
    setMutating(true)

    const stepNumber = parseInt(data.stepNumber)

    const { name, description } = data

    const variables = {
      name,
      slug,
      stepNumber,
      description,
      useCaseId
    }
    updateUseCaseStep({
      variables,
      context: {
        headers: {
          'Accept-Language': locale
        }
      }
    })
  }

  const cancelForm = () => {
    setReverting(true)
    router.push(
      `/${locale}` +
      `/use-cases/${useCase?.slug}` +
      `/${slug && 'use-case-steps/' + slug}`
    )
  }

  return (
    <form onSubmit={handleSubmit(doUpsert)}>
      <div className='px-4 lg:px-0 py-4 lg:py-6'>
        <div className='flex flex-col gap-y-6 text-sm'>
          <div className='text-2xl font-semibold text-dial-sapphire'>
            {useCaseStep?.slug
              ? format('app.editEntity', { entity: useCaseStep?.name })
              : `${format('app.createNew')} ${format('use-case-step.label')}`
            }
          </div>
          <div className='form-field-wrapper'>
            <label className='required-field text-dial-blueberry' htmlFor='name'>
              {format('use-case-step.name')}
            </label>
            <Input
              {...register('name', { required: format('validation.required') })}
              id='name'
              placeholder={format('use-case-step.name')}
              isInvalid={errors.name}
            />
            {errors.name && <ValidationError value={errors.name?.message} />}
          </div>
          <div className='form-field-wrapper'>
            <label className='required-field text-dial-blueberry' htmlFor='stepNumber'>
              {format('use-case-step.step-number')}
            </label>
            <Input
              type='number'
              {...register('stepNumber', { required: format('validation.required') })}
              id='stepNumber'
              placeholder={format('use-case-step.step-number')}
              isInvalid={errors.stepNumber}
            />
            {errors.stepNumber && <ValidationError value={errors.stepNumber?.message} />}
          </div>
          <div className='flex flex-col gap-y-2'>
            <label id='description' className='required-field text-dial-blueberry'>
              {format('use-case-step.description')}
            </label>
            <Controller
              name='description'
              control={control}
              render={({ field: { value, onChange } }) => (
                <HtmlEditor
                  labelledBy='description'
                  onChange={onChange}
                  initialContent={value}
                  placeholder={format('use-case-step.description')}
                />
              )}
              rules={{ required: format('validation.required') }}
            />
            {errors.description && <ValidationError value={errors.description?.message} />}
          </div>
          <div className='flex flex-wrap text-base mt-6 gap-3'>
            <button
              type='submit'
              className='submit-button'
              disabled={mutating || reverting}
            >
              {`${format('app.submit')} ${format('use-case-step.label')}`}
              {mutating && <FaSpinner className='spinner ml-3' />}
            </button>
            <button
              type='button'
              className='cancel-button'
              disabled={mutating || reverting}
              onClick={cancelForm}
            >
              {format('app.cancel')}
              {reverting && <FaSpinner className='spinner ml-3' />}
            </button>
          </div>
          {useCase.markdownUrl &&
            <div className='text-sm italic text-red-500 -mt-3'>
              {format('useCaseStep.markdownWarning')}
            </div>
          }
        </div>
      </div>
    </form>
  )
})

UseCaseStepForm.displayName = 'UseCaseStepForm'

export default UseCaseStepForm
