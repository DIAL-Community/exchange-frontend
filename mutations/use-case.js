import { gql } from '@apollo/client'

export const CREATE_USE_CASE_STEP = gql`
  mutation CreateUseCaseStep(
    $name: String!,
    $slug: String!,
    $stepNumber: Int!,
    $description: String
    $useCaseId: Int!
    $markdownUrl: String
  ) {
    createUseCaseStep(
      name: $name
      slug: $slug
      stepNumber: $stepNumber
      description: $description
      useCaseId: $useCaseId
      markdownUrl: $markdownUrl
    ) {
      useCaseStep {
        slug
        useCase {
          slug
        }
      },
      errors
    }
  }
`

export const UPDATE_USE_CASE_SDG_TARGETS = gql`
  mutation UpdateUseCaseSdgTargets (
    $sdgTargetsIds: [Int!]!
    $slug: String!
  ) {
    updateUseCaseSdgTargets (
      sdgTargetsIds: $sdgTargetsIds
      slug: $slug
    ) {
      useCase {
        slug
        sdgTargets {
          id
          name
          targetNumber
          sustainableDevelopmentGoal {
           slug
          }
        }
      }
      errors
    }
  }
`

export const CREATE_USE_CASE = gql`
  mutation CreateUseCase(
    $name: String!
    $slug: String!
    $sectorSlug: String!
    $maturity: String!
    $imageFile: Upload
    $description: String!
  ) {
    createUseCase(
      name: $name
      slug: $slug
      sectorSlug: $sectorSlug
      maturity: $maturity
      imageFile: $imageFile
      description: $description
    ) {
      useCase {
        slug
      }
      errors
    }
  }
`

export const UPDATE_USE_CASE_TAGS = gql`
  mutation UpdateUseCaseTags(
    $slug: String!
    $tags: [String!]!
  ) {
    updateUseCaseTags(
      slug: $slug
      tags: $tags
    ) {
      useCase {
        tags
      }
      errors
    }  
  }
`
