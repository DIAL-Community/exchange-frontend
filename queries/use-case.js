import { gql } from '@apollo/client'

export const USE_CASE_QUERY = gql`
  query UseCase($slug: String!) {
    useCase(slug: $slug) {
      id
      name
      slug
      maturity
      sector {
        slug
        name
        id
      }
      useCaseDescription {
        description
      }
    }
  }
`

export const USE_CASE_DETAIL_QUERY = gql`
  query UseCase($slug: String!) {
    useCase(slug: $slug) {
      id
      name
      slug
      sector {
        slug
        name
        id
      }
      maturity
      useCaseDescription {
        description
      }
      imageFile
      useCaseDescription {
        description
        locale
      }
      sdgTargets {
        id
        name
        targetNumber
        sustainableDevelopmentGoal {
          slug
        }
      }
      workflows {
        name
        slug
        imageFile
      }
      buildingBlocks {
        name
        slug
        maturity
        imageFile
      }
      useCaseHeaders {
        header
      }
      tags
    }
  }
`

export const USE_CASE_SEARCH_QUERY = gql`
  query UseCases($search: String!, $mature: Boolean!) {
    useCases(search: $search, mature: $mature) {
      id
      name
      slug
    }
  }
`
