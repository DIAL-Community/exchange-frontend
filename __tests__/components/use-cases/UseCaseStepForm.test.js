import { act } from 'react-dom/test-utils'
import { fireEvent, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { render, waitForAllEffects, waitForAllEffectsAndSelectToLoad } from '../../test-utils'
import StepForm from '../../../components/use-cases/steps/StepForm'
import { CREATE_USE_CASE_STEP } from '../../../mutations/use-case'
import { mockNextAuthUseSession, mockNextUseRouter, statuses } from '../../utils/nextMockImplementation'
import { createUseCaseStepSuccess, useCaseStep, useCase } from './data/UseCaseStepForm'

mockNextUseRouter()
describe('Unit tests for UseCaseStepForm component.', () => {
  const USE_CASE_STEP_NAME_TEST_ID = 'use-case-step-name'
  const USE_CASE_STEP_DESCRIPTION_TEST_ID = 'use-case-step-description'
  const USE_CASE_STEP_STEP_NUMBER_TEST_ID = 'use-case-step-step-number'
  const SUBMIT_BUTTON_TEST_ID = 'submit-button'
  const REQUIRED_FIELD_MESSAGE = 'This field is required'
  const mockCreateUseCaseStepVariables = {
    name: 'Test Use Case Step',
    slug: 'test_use_case_step',
    stepNumber: 1,
    description: 'test Use Case Step description',
    useCaseId: 17
  }

  test('Should render Unauthorized component for unauthorized user.', async () => {
    mockNextAuthUseSession(statuses.UNAUTHENTICATED)
    const { container } = render(
      <CustomMockedProvider>
        <StepForm useCase={useCase} />
      </CustomMockedProvider>
    )
    await waitForAllEffects()
    expect(container).toHaveTextContent('You are not authorized to view this page')
  })

  test('Should render Unauthorized component for user who is not an admin.', async () => {
    mockNextAuthUseSession(statuses.AUTHENTICATED, { isAdminUser: false })
    const { container } = render(
      <CustomMockedProvider>
        <StepForm useCase={useCase} />
      </CustomMockedProvider>
    )
    await waitForAllEffects()
    expect(container).toHaveTextContent('You are not authorized to view this page')
  })

  test('Should render StepForm component for admin user.', async () => {
    mockNextAuthUseSession(statuses.AUTHENTICATED, { isAdminUser: true })
    const { container } = render(
      <CustomMockedProvider>
        <StepForm useCase={useCase} />
      </CustomMockedProvider>
    )
    await waitForAllEffects()
    expect(container).toMatchSnapshot()
  })

  test('Should show validation errors for mandatory fields and hide them on input value change.', async () => {
    const user = userEvent.setup()
    mockNextAuthUseSession(statuses.AUTHENTICATED, { isAdminUser: true })
    const { getByTestId, container } = render(
      <CustomMockedProvider>
        <StepForm useCase={useCase} />
      </CustomMockedProvider>
    )
    await waitForAllEffectsAndSelectToLoad(container)
    await act(() => fireEvent.submit(getByTestId(SUBMIT_BUTTON_TEST_ID)))

    expect(getByTestId(USE_CASE_STEP_NAME_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(USE_CASE_STEP_DESCRIPTION_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(USE_CASE_STEP_STEP_NUMBER_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)

    await user.type(screen.getByLabelText(/Name/), 'test use case step name')
    expect(getByTestId(USE_CASE_STEP_NAME_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    await user.clear(screen.getByLabelText(/Name/))
    expect(getByTestId(USE_CASE_STEP_NAME_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)

    await user.type(screen.getByLabelText(/Name/), 'test use case step name 2')
    expect(getByTestId(USE_CASE_STEP_NAME_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(USE_CASE_STEP_DESCRIPTION_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(USE_CASE_STEP_STEP_NUMBER_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)

    await user.type(screen.getByLabelText(/Step Number/), '1')
    expect(getByTestId(USE_CASE_STEP_STEP_NUMBER_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    await user.clear(screen.getByLabelText(/Step Number/))
    expect(getByTestId(USE_CASE_STEP_STEP_NUMBER_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)

    await user.type(screen.getByLabelText(/Step Number/), '2')
    expect(getByTestId(USE_CASE_STEP_STEP_NUMBER_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(USE_CASE_STEP_DESCRIPTION_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(USE_CASE_STEP_NAME_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)

    await act(() => fireEvent.submit(getByTestId(SUBMIT_BUTTON_TEST_ID)))
    expect(getByTestId(USE_CASE_STEP_NAME_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(USE_CASE_STEP_DESCRIPTION_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    expect(getByTestId(USE_CASE_STEP_STEP_NUMBER_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
  })

  test('Should display success toast on submit.', async () => {
    mockNextAuthUseSession(statuses.AUTHENTICATED, { isAdminUser: true })
    const mockCreateUseCaseStep = generateMockApolloData(
      CREATE_USE_CASE_STEP,
      mockCreateUseCaseStepVariables,
      null,
      createUseCaseStepSuccess
    )
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockCreateUseCaseStep]}>
        <StepForm useCaseStep={useCaseStep} useCase={useCase}/>
      </CustomMockedProvider>
    )
    await waitForAllEffects()

    fireEvent.submit(getByTestId(SUBMIT_BUTTON_TEST_ID))
    await screen.findByText('Use Case Step submitted successfully')
    expect(container).toMatchSnapshot()
  })

  test('Should display failure toast on submit.', async () => {
    mockNextAuthUseSession(statuses.AUTHENTICATED, { isAdminUser: true })
    const errorMessage = 'An error occurred'
    const mockCreateUseCaseStep = generateMockApolloData(
      CREATE_USE_CASE_STEP,
      mockCreateUseCaseStepVariables,
      new Error(errorMessage)
    )
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockCreateUseCaseStep]}>
        <StepForm useCaseStep={useCaseStep} useCase={useCase}/>
      </CustomMockedProvider>
    )
    await waitForAllEffects()

    fireEvent.submit(getByTestId(SUBMIT_BUTTON_TEST_ID))
    await screen.findByText('Use Case Step submission failed')
    await screen.findByText(errorMessage)
    expect(container).toMatchSnapshot()
  })
})
