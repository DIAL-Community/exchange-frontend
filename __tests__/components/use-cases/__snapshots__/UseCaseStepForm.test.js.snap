// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for UseCaseStepForm component. Should display failure toast on submit. 1`] = `
<div>
  <div
    class="Toastify"
  >
    <div
      class="Toastify__toast-container Toastify__toast-container--top-center"
      style="width: auto;"
    >
      <div
        class="Toastify__toast Toastify__toast-theme--light Toastify__toast--error Toastify__toast--close-on-click Toastify--animate Toastify__slide-enter--top-center"
        id="toastId"
        style="--nth: 1; --len: 1;"
      >
        <div
          class="Toastify__toast-body"
          role="alert"
        >
          <div
            class="Toastify__toast-icon Toastify--animate-icon Toastify__zoom-enter"
          >
            <svg
              fill="var(--toastify-icon-color-error)"
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
            >
              <path
                d="M11.983 0a12.206 12.206 0 00-8.51 3.653A11.8 11.8 0 000 12.207 11.779 11.779 0 0011.8 24h.214A12.111 12.111 0 0024 11.791 11.766 11.766 0 0011.983 0zM10.5 16.542a1.476 1.476 0 011.449-1.53h.027a1.527 1.527 0 011.523 1.47 1.475 1.475 0 01-1.449 1.53h-.027a1.529 1.529 0 01-1.523-1.47zM11 12.5v-6a1 1 0 012 0v6a1 1 0 11-2 0z"
              />
            </svg>
          </div>
          <div>
            <div
              class="flex flex-col"
            >
              <span>
                Use Case Step submission failed
              </span>
              <span>
                An error occurred
              </span>
            </div>
          </div>
        </div>
        <button
          aria-label="close"
          class="Toastify__close-button Toastify__close-button--light"
          type="button"
        >
          <svg
            aria-hidden="true"
            viewBox="0 0 14 16"
          >
            <path
              d="M7.71 8.23l3.75 3.75-1.48 1.48-3.75-3.75-3.75 3.75L1 11.98l3.75-3.75L1 4.48 2.48 3l3.75 3.75L9.98 3l1.48 1.48-3.75 3.75z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <div
          aria-hidden="true"
          aria-label="notification timer"
          class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar-theme--light Toastify__progress-bar--error"
          role="progressbar"
          style="animation-duration: 1000ms; animation-play-state: paused; opacity: 0;"
        />
      </div>
    </div>
  </div>
  <div
    class="flex flex-col"
  >
    <div
      class="hidden lg:block px-8"
    >
      <div
        class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="inline text-dial-sapphire h5"
          href="/"
        >
          Home
        </a>
        <div
          class="inline h5"
        >
           &gt; 
          <a
            class="text-dial-gray-dark"
            href="/"
          />
        </div>
      </div>
    </div>
    <div
      class="pb-8 px-8"
    >
      <div
        class="sm:px-0 max-w-full mx-auto"
        id="content"
      >
        <form>
          <div
            class="bg-edit shadow-md rounded px-8 pt-6 pb-12 mb-4 flex flex-col gap-3"
          >
            <div
              class="text-2xl font-semibold text-dial-sapphire pb-4"
            >
              Edit Test Use Case Step
            </div>
            <div
              class="flex flex-col lg:flex-row gap-4"
            >
              <div
                class="w-full lg:w-1/2 flex flex-col gap-y-3"
              >
                <div
                  class="form-field-wrapper"
                  data-testid="use-case-step-name"
                >
                  <label
                    class="form-field-label required-field"
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    class="w-full"
                    id="name"
                    name="name"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="use-case-step-step-number"
                >
                  <label
                    class="form-field-label required-field"
                    for="stepNumber"
                  >
                    Step Number
                  </label>
                  <input
                    class="w-full"
                    id="stepNumber"
                    name="stepNumber"
                    placeholder="Step Number"
                    type="number"
                    value=""
                  />
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="use-case-step-markdown-url"
                >
                  <label
                    class="form-field-label required-field"
                    for="markdownUrl"
                  >
                    Markdown Filename
                  </label>
                  <input
                    class="w-full"
                    id="markdownUrl"
                    name="markdownUrl"
                    placeholder="Markdown Filename"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="w-full lg:w-1/2"
              >
                <div
                  class="block flex flex-col gap-y-2"
                  data-testid="use-case-step-description"
                >
                  <label
                    class="form-field-label required-field"
                  >
                    Description
                  </label>
                  <div
                    class="relative"
                  >
                    <svg
                      class="absolute text-lg inset-x-1/2 top-10 spinner"
                      fill="currentColor"
                      height="2em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="2em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                      />
                    </svg>
                    <div
                      class="html-editor"
                    >
                      <textarea
                        id="description-editor"
                        style="visibility: hidden;"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-wrap text-xl mt-8 gap-3"
            >
              <button
                class="submit-button"
                data-testid="submit-button"
                type="submit"
              >
                Submit Use Case Step
              </button>
              <button
                class="cancel-button"
                type="button"
              >
                Cancel
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for UseCaseStepForm component. Should display success toast on submit. 1`] = `
<div>
  <div
    class="Toastify"
  >
    <div
      class="Toastify__toast-container Toastify__toast-container--top-center"
      style="width: auto;"
    >
      <div
        class="Toastify__toast Toastify__toast-theme--light Toastify__toast--success Toastify__toast--close-on-click Toastify--animate Toastify__slide-enter--top-center"
        id="toastId"
        style="--nth: 1; --len: 1;"
      >
        <div
          class="Toastify__toast-body"
          role="alert"
        >
          <div
            class="Toastify__toast-icon Toastify--animate-icon Toastify__zoom-enter"
          >
            <svg
              fill="var(--toastify-icon-color-success)"
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
            >
              <path
                d="M12 0a12 12 0 1012 12A12.014 12.014 0 0012 0zm6.927 8.2l-6.845 9.289a1.011 1.011 0 01-1.43.188l-4.888-3.908a1 1 0 111.25-1.562l4.076 3.261 6.227-8.451a1 1 0 111.61 1.183z"
              />
            </svg>
          </div>
          <div>
            Use Case Step submitted successfully
          </div>
        </div>
        <button
          aria-label="close"
          class="Toastify__close-button Toastify__close-button--light"
          type="button"
        >
          <svg
            aria-hidden="true"
            viewBox="0 0 14 16"
          >
            <path
              d="M7.71 8.23l3.75 3.75-1.48 1.48-3.75-3.75-3.75 3.75L1 11.98l3.75-3.75L1 4.48 2.48 3l3.75 3.75L9.98 3l1.48 1.48-3.75 3.75z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <div
          aria-hidden="true"
          aria-label="notification timer"
          class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar-theme--light Toastify__progress-bar--success"
          role="progressbar"
          style="animation-duration: 1000ms; animation-play-state: paused; opacity: 0;"
        />
      </div>
    </div>
  </div>
  <div
    class="flex flex-col"
  >
    <div
      class="hidden lg:block px-8"
    >
      <div
        class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="inline text-dial-sapphire h5"
          href="/"
        >
          Home
        </a>
        <div
          class="inline h5"
        >
           &gt; 
          <a
            class="text-dial-gray-dark"
            href="/"
          />
        </div>
      </div>
    </div>
    <div
      class="pb-8 px-8"
    >
      <div
        class="sm:px-0 max-w-full mx-auto"
        id="content"
      >
        <form>
          <div
            class="bg-edit shadow-md rounded px-8 pt-6 pb-12 mb-4 flex flex-col gap-3"
          >
            <div
              class="text-2xl font-semibold text-dial-sapphire pb-4"
            >
              Edit Test Use Case Step
            </div>
            <div
              class="flex flex-col lg:flex-row gap-4"
            >
              <div
                class="w-full lg:w-1/2 flex flex-col gap-y-3"
              >
                <div
                  class="form-field-wrapper"
                  data-testid="use-case-step-name"
                >
                  <label
                    class="form-field-label required-field"
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    class="w-full"
                    id="name"
                    name="name"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="use-case-step-step-number"
                >
                  <label
                    class="form-field-label required-field"
                    for="stepNumber"
                  >
                    Step Number
                  </label>
                  <input
                    class="w-full"
                    id="stepNumber"
                    name="stepNumber"
                    placeholder="Step Number"
                    type="number"
                    value=""
                  />
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="use-case-step-markdown-url"
                >
                  <label
                    class="form-field-label required-field"
                    for="markdownUrl"
                  >
                    Markdown Filename
                  </label>
                  <input
                    class="w-full"
                    id="markdownUrl"
                    name="markdownUrl"
                    placeholder="Markdown Filename"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="w-full lg:w-1/2"
              >
                <div
                  class="block flex flex-col gap-y-2"
                  data-testid="use-case-step-description"
                >
                  <label
                    class="form-field-label required-field"
                  >
                    Description
                  </label>
                  <div
                    class="relative"
                  >
                    <svg
                      class="absolute text-lg inset-x-1/2 top-10 spinner"
                      fill="currentColor"
                      height="2em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="2em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                      />
                    </svg>
                    <div
                      class="html-editor"
                    >
                      <textarea
                        id="description-editor"
                        style="visibility: hidden;"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-wrap text-xl mt-8 gap-3"
            >
              <button
                class="submit-button"
                data-testid="submit-button"
                type="submit"
              >
                Submit Use Case Step
              </button>
              <button
                class="cancel-button"
                type="button"
              >
                Cancel
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for UseCaseStepForm component. Should render StepForm component for admin user. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="flex flex-col"
  >
    <div
      class="hidden lg:block px-8"
    >
      <div
        class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="inline text-dial-sapphire h5"
          href="/"
        >
          Home
        </a>
        <div
          class="inline h5"
        >
           &gt; 
          <a
            class="text-dial-gray-dark"
            href="/"
          />
        </div>
      </div>
    </div>
    <div
      class="pb-8 px-8"
    >
      <div
        class="sm:px-0 max-w-full mx-auto"
        id="content"
      >
        <form>
          <div
            class="bg-edit shadow-md rounded px-8 pt-6 pb-12 mb-4 flex flex-col gap-3"
          >
            <div
              class="text-2xl font-semibold text-dial-sapphire pb-4"
            >
              Create New Use Case Step
            </div>
            <div
              class="flex flex-col lg:flex-row gap-4"
            >
              <div
                class="w-full lg:w-1/2 flex flex-col gap-y-3"
              >
                <div
                  class="form-field-wrapper"
                  data-testid="use-case-step-name"
                >
                  <label
                    class="form-field-label required-field"
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    class="w-full"
                    id="name"
                    name="name"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="use-case-step-step-number"
                >
                  <label
                    class="form-field-label required-field"
                    for="stepNumber"
                  >
                    Step Number
                  </label>
                  <input
                    class="w-full"
                    id="stepNumber"
                    name="stepNumber"
                    placeholder="Step Number"
                    type="number"
                    value=""
                  />
                </div>
                <div
                  class="form-field-wrapper"
                  data-testid="use-case-step-markdown-url"
                >
                  <label
                    class="form-field-label required-field"
                    for="markdownUrl"
                  >
                    Markdown Filename
                  </label>
                  <input
                    class="w-full"
                    id="markdownUrl"
                    name="markdownUrl"
                    placeholder="Markdown Filename"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="w-full lg:w-1/2"
              >
                <div
                  class="block flex flex-col gap-y-2"
                  data-testid="use-case-step-description"
                >
                  <label
                    class="form-field-label required-field"
                  >
                    Description
                  </label>
                  <div
                    class="relative"
                  >
                    <svg
                      class="absolute text-lg inset-x-1/2 top-10 spinner"
                      fill="currentColor"
                      height="2em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="2em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                      />
                    </svg>
                    <div
                      class="html-editor"
                    >
                      <textarea
                        id="description-editor"
                        style="visibility: hidden;"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-wrap text-xl mt-8 gap-3"
            >
              <button
                class="submit-button"
                data-testid="submit-button"
                type="submit"
              >
                Submit Use Case Step
              </button>
              <button
                class="cancel-button"
                type="button"
              >
                Cancel
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
